### PR TITLE
Fix handling of WS Close frame in Netty backends

### PR DESCRIPTION
**Pull request by kciesielski**
Fixes https://github.com/softwaremill/tapir/issues/3776

With this fix, netty-cats and netty-sync backends should handle `Close` frame sequentially, after all previously received frames are processed.

- [x] depends on https://github.com/softwaremill/ox/pull/151

### CODE REVIEW

File: server/netty-server/cats/src/main/scala/sttp/tapir/server/netty/cats/internal/WebSocketPipeProcessor.scala [hunk 1]
```diff
@@ -98,10 +100,19 @@ class WebSocketPipeProcessor[F[_]: Async, REQ, RESP](
...
       s.mapAccumulate(None: Accumulator)(accumulateFrameState).collect { case (_, Some(f)) => f }
     } else s
+
+  private def takeUntilCloseFrame(passAlongCloseFrame: Boolean)(s: Stream[F, WebSocketFrame]): Stream[F, WebSocketFrame] =
+    s.takeWhile(
+      {
+        case _: WebSocketFrame.Close => false
+        case _                       => true
+      },
+      takeFailure = passAlongCloseFrame
```
Comment (adamw) [inline server/netty-server/cats/src/main/scala/sttp/tapir/server/netty/cats/internal/WebSocketPipeProcessor.scala:114]: hm this (`takeFailure`) looks weird - we should find a better name ;) this is only for the *first* frame that fails the test, right?

**Review by adamw — APPROVED** [review]
Nice! :)

File: server/netty-server/cats/src/main/scala/sttp/tapir/server/netty/cats/internal/WebSocketPipeProcessor.scala (hunk 1 above)
Comment (kciesielski) [reply server/netty-server/cats/src/main/scala/sttp/tapir/server/netty/cats/internal/WebSocketPipeProcessor.scala:114]: Yes. It's how they called it in `fs2`. My first version was named `takeLast`, but I wasn't happy with it (takeLast = false looks like it skips the failing element, so takes even less?). ZIO has a `takeUntil` method to deal with this.
Akka streams has `inclusive: Boolean` and this looks better than `takeFailure`, wdyt?

Comment (adamw) [reply server/netty-server/cats/src/main/scala/sttp/tapir/server/netty/cats/internal/WebSocketPipeProcessor.scala:114]: `includeFirstFailing`?

**Pull request merged**